src/main/scala/mesosphere/marathon/MarathonSchedulerService.scala

```diff
@@ -40,12 +40,11 @@ class MarathonSchedulerService @Inject() (
     healthCheckManager: HealthCheckManager,
     @Named(ModuleNames.NAMED_CANDIDATE) candidate: Option[Candidate],
     config: MarathonConf,
-    httpConfig: HttpConf,
     frameworkIdUtil: FrameworkIdUtil,
     @Named(ModuleNames.NAMED_LEADER_ATOMIC_BOOLEAN) leader: AtomicBoolean,
     appRepository: AppRepository,
     taskTracker: TaskTracker,
-    scheduler: MarathonScheduler,
+    newDriver: Provider[SchedulerDriver],
```

Comment:
Probably the same topic - sorry: why ingesting the provider (which is a javax.inject class)?